backend/src/api/open_source_project.ts

```diff
@@ -0,0 +1,263 @@
+ *    tags:
+ *    - OpenSourceProjects
+ */
+export const getById = wrapHandler(async (event) => {
+  const id = event.pathParameters?.projectId;
+  if (!id) {
+    return {
+      statusCode: 400,
+      body: JSON.stringify({ error: 'Project ID is required' }),
+    };
+  }
+ 
+  await connectToDatabase();
+
+  const project = await OpenSourceProject.findOne(id, {
+    relations: ['organizations'],
+  });
+
+  if (!project) {
+    return {
+      statusCode: 404,
+      body: JSON.stringify({ error: 'Project not found' }),
+    };
+  }
+ 
+  // retrieve associated orgid(s)
+  const orgIds = project.organizations.map(org => org.id);
+  const userOrgs = getOrgMemberships(event);
+  
+  if (!isGlobalViewAdmin(event) && !isGlobalWriteAdmin(event) && !userOrgs.some(userId => orgIds.includes(userId))) return Unauthorized;
+  
+  return {
+    
+    statusCode: 200,
+    body: JSON.stringify(project),
+  };
+});
+
+
+/**
+ * @swagger
+ *
+ * /project/listOrgs/{orgId}:
+ *  list:
+ *    description: List open source projects for given organization.
+ *    parameters:
+ *      - in: path
+ *        name: orgId
+ *        required: true
+ *        description: Organization ID
+ *    tags:
+ *    - Organizations
+ */
+export const listByOrg = wrapHandler(async (event) => {
+  console.log("im here");
+  const orgId = event.pathParameters?.orgId;
+  // check permissions
+  if (!orgId || (!isGlobalViewAdmin(event) && !isGlobalWriteAdmin(event) && !getOrgMemberships(event).includes(orgId))) return Unauthorized;
+  await connectToDatabase();
+  console.log("made it here");
+  const organization = await Organization.findOne({
+    where: { id: orgId },
+    relations: ['openSourceProjects']
+  });
+  console.log("my org = ", organization);
+  if (organization) {
+    return {
+      statusCode: 200,
+      body: JSON.stringify(organization.openSourceProjects)
+    };
+  }
+  return NotFound;
+});
+
+
+
+/**
+ * @swagger
+ *
+ * /project_upsert/org/{orgId}:
+ *  create:
+ *    description: Create a new open source project without an organization if it doesn't already exist and add it to a given organization.
+ *    tags:
+ *    - Organizations
+ */
+export const create_proj = wrapHandler(async (event) => {
+  const orgId = event.pathParameters?.orgId;
+  const parsedBody = JSON.parse(event.body ?? '{}');
+  const { url, hipcheckResults, organizations } = parsedBody;
+
+  // check permissions
+  if (!orgId || (!isGlobalWriteAdmin(event) && !getOrgMemberships(event).includes(orgId))) return Unauthorized;
+  
+  // const body = await validateBody(OpenSourceProject, event.body);
+  await connectToDatabase();
+
+  let openSourceProject: OpenSourceProject | undefined;
+
+  // Check if there's an existing open source project without an organization
+  const existingProjects = await OpenSourceProject.createQueryBuilder('osp')
+    .leftJoinAndSelect('osp.organizations', 'organizations')
+    .where('organizations.id IS NULL')
```

Comment:
I think this would always not find any open source project since we delete a project if it has no organization. Instead, could we query for open source projects with the given url and if there is one, attach it to the organization, and if not create a new one?